frontend/src/components/SelectColumnValidator/utils/index.ts

```diff
@@ -0,0 +1,116 @@
+interface ParsedOptions {
+  id: string;
+  values: any[];
+}
+
+export const validateSelectColumnDeselect = (options: string, checkboxValue: string) => {
+  return new Promise((resolve: (validationMessage: string) => void) => {
+    const parsedOptions = parseOptions(options);
+    parsedOptions?.values?.flat().reduce(function (result, query) {
+      if (query && query.field === checkboxValue && parsedOptions.id === 'filters') {
+        resolve(
+          `Notice that ${query.field
+            .split('_')
+            .join(' ')} is used in a filter step, deselecting it here may break this query.`,
```

Comment:
Given the repetitive occurrences, extract this into a variable and keep switching out the dynamic bits i.e. field & step (filter, aggregate, e.t.c.)

Reply:
Now using string literals